packages/amplify-migration-template-gen/src/category-template-generator.ts

```diff
@@ -52,6 +63,27 @@ class CategoryTemplateGenerator<CFNCategoryType extends CFN_CATEGORY_TYPE> {
     );
     const gen1TemplateWithDepsResolved = new CfnDependencyResolver(gen1TemplateWithOutputsResolved).resolve(logicalResourceIds);
     const gen1TemplateWithConditionsResolved = new CFNConditionResolver(gen1TemplateWithDepsResolved).resolve(Parameters);
+    const oAuthProvidersParam = Parameters.find((param) => param.ParameterKey === HOSTED_PROVIDER_META_PARAMETER_NAME);
+    if (oAuthProvidersParam) {
+      const userPoolId = Outputs.find((op) => op.OutputKey === USER_POOL_ID_OUTPUT_KEY_NAME)?.OutputValue;
+      assert(userPoolId);
+      const oAuthValues = await retrieveOAuthValues({
+        ssmClient: this.ssmClient,
+        cognitoIdpClient: this.cognitoIdpClient,
+        appId: this.appId,
+        environmentName: this.environmentName,
+        oAuthParameter: oAuthProvidersParam,
+        userPoolId,
+      });
+      const oAuthProviderCredentialsParam = Parameters.find((param) => param.ParameterKey === HOSTED_PROVIDER_CREDENTIALS_PARAMETER_NAME);
+      assert(oAuthProviderCredentialsParam);
+      oAuthProviderCredentialsParam.ParameterValue = JSON.stringify(oAuthValues);
+      return {
```

Comment:
Do we need the return statement here? Since we already have the same return in line 87?